Add Beat config library (#3406)
elastic/cloud-on-k8s: config/recipes/beats/README.md

> # Beats Configuration Library

Reviewer: Can "library" be a confusing term to use? To me it sounds more like a programmatic thing than a collection of examples. I'd appreciate thought from better english speakers though (cc. @anyasabo @charith-elastic)

Reviewer 2: Good point! Maybe we should just go with "Beats Configuration Examples"?

Reviewer 3: _The Beats Zoo_?  :stuck_out_tongue: 
+1 for _Beats Configuration Examples_

Reviewer 4: 👍 Examples or Cookbook I think both avoid the ambiguity seb mentioned

> # Beats Configuration Library

Reviewer: I'm wondering how those examples will be referenced from the documentation on the website. I think there's value in listing all the examples explicitly in the docs (ie. copying this markdown file content, basically).

Reviewer 3: [I tried to include the samples in the documentation a long time ago](https://github.com/elastic/cloud-on-k8s/pull/2656) but the docs build only checks out the `docs` directory so that wasn't successful. We'll either have to move the samples inside the docs directory or figure out the magic incantation required by the docs build to checkout more directories.

Author: I planned to copy it to the docs almost 1:1.

> This directory contains yaml manifests with example configurations for Beats. These manifests are self-contained and work out-of-the-box on any non-secured Kubernetes cluster. All of them contain three-node Elasticsearch cluster and single Kibana instance. All Beat configurations set up Kibana dashboards if they are available for a given Beat and all required RBAC resources. 

Reviewer 4: Should we add a note that none of these set resource requests/limits and they may want to set these in a production deployment?

Reviewer: > none of these set resource requests/limits

should we set default reasonable ones in the examples directly?

Author: > > none of these set resource requests/limits
> 
> should we set default reasonable ones in the examples directly?

For Beats, I'd say our defaults are already reasonable. I think we shouldn't overload those examples.

Reviewer 4: Ah I wondered because even on a cluster with very little traffic packetbeat was using up a lot of its CPU, but I'm good with not mentioning it

> This directory contains yaml manifests with example configurations for Beats. These manifests are self-contained and work out-of-the-box on any non-secured Kubernetes cluster. All of them contain three-node Elasticsearch cluster and single Kibana instance. All Beat configurations set up Kibana dashboards if they are available for a given Beat and all required RBAC resources. 

Reviewer: Should we mention how the Elasticsearch spec is not suitable to a production environment?

Author: This is already there in https://github.com/elastic/cloud-on-k8s/blob/master/config/recipes/README.asciidoc, should we just repeat it here?

Reviewer: I'm not sure people will notice the root README, it's worth repeating here imo.

#### Metricbeat for Kubernetes monitoring - `metricbeat_hosts.yaml`

> Deploys Metricbeat as a DaemonSet that monitors the host resource usage (cpu, memory, network, filesystem) and Kubernetes resources (nodes, pods, containers, volumes).

Reviewer 4: ```suggestion
Deploys Metricbeat as a DaemonSet that monitors the host resource usage (cpu, memory, network, filesystem) and Kubernetes resources (Nodes, Pods, Contains, Volumes).
```

#### Filebeat with autodiscover - `filebeat_autodiscover.yaml`

> Deploys Filebeat as DaemonSet with autodiscover feature enabled. All pods in all namespace will have logs shipped to a Elasticsearch cluster.

Reviewer 4: ```suggestion
Deploys Filebeat as DaemonSet with autodiscover feature enabled. All pods in all namespaces will have logs shipped to a Elasticsearch cluster.
```

#### Filebeat with autodiscover for metadata - `filebeat_autodiscover_by_metadata.yaml`

> Deploys Filebeat as a DaemonSet with autodiscover feature enabled. Fullfilling any of the two conditions below will cause a given Pod logs to be shipped to a Elasticsearch cluster:

Reviewer 4: ```suggestion
Deploys Filebeat as a DaemonSet with autodiscover feature enabled. Fullfilling any of the two conditions below will cause a given Pod's logs to be shipped to an Elasticsearch cluster:
```

- Pod is in `log-namespace` namespace
- Pod has `log-label: "true"` label 

#### Filebeat without autodiscover - `filebeat_no_autodiscover.yaml`

Deploys Filebeat as a DaemonSet with autodiscover feature disabled. Uses entire logs directory on the host as the input. Doesn't require any RBAC resources as no Kubernetes APIs are used.   

#### Heartbeat monitoring Elasticsearch and Kibana health - `heartbeat_es_kb_heatlh.yaml`

Deploys Heartbeat as a single Pod deployment that monitors the health of Elasticsearch and Kibana by TCP probing their Service endpoints.

#### Auditbeat - `auditbeat_hosts.yaml`

Deploys Auditbeat as a DaemonSet that checks file integrity and audits file operations on the host system.

#### Journalbeat - `journalbeat_hosts.yaml`

Deploys Journalbeat as a DaemonSet that ships data from systemd journals.

> 

Reviewer: nit: new line
